clinica/converters/_utils.py

```diff
@@ -175,33 +183,67 @@ def create_participants_df(
         participant_df = participant_df.drop_duplicates(
             subset=["alternative_id_1"], keep="first"
         )
+
     elif study_name == StudyName.OASIS:
         # OASIS provides several MRI for the same session
         participant_df = participant_df[
             ~participant_df.alternative_id_1.str.endswith("_MR2")
         ]
+
     participant_df.reset_index(inplace=True, drop=True)
 
+    # Extracting subjects list from the input subjects list file
+    if subjects and subjects_pattern:
+        import re
+
+        rgx = re.compile(subjects_pattern)
+        subjects_list = filter(rgx.fullmatch, get_subjects_list_from_file(subjects))
```

Comment:
Here we handle the `subjects` parameter of the function:

```python
if subjects:
    subjects_list = [
        bids_id_factory(study_name).from_original_study_id(subject)
        for subject in get_subjects_list_from_file(subjects)
    ]
```

Then, we combine `bids_ids` and `subjects_list` into a single filtering set:

```python
bids_id_to_keep: set[BIDSSubjectID] = set(bids_ids + subjects_list)
```